### DISCUSSION TITLE
when using prettier

### DISCUSSION THREAD

**Discussion by JuanDavila1101** [nss-evening-cohort-14/evening-client-side#139] [Help Me!] [answered]
when using prettier Ctrl +Shift + F is adding extra lines to the code in unwanted places 

this is correct 
![image](https://user-images.githubusercontent.com/76821980/108940979-e13c0880-7619-11eb-985b-f14d15e0daa9.png)

after prettier
![image](https://user-images.githubusercontent.com/76821980/108941052-029cf480-761a-11eb-954a-7dde13a564dc.png)

I have looked at my settings here:
https://github.com/nss-nightclass-projects/Night-Class-Resources/blob/master/book-1-foundations/chapters/vs_code.md#settings-to-change

these are the new lines and indentation settings on the Visual Studio Code 
![image](https://user-images.githubusercontent.com/76821980/108940504-16942680-7619-11eb-8c3d-ab09ea43d8e5.png)

similar issue here:
https://github.com/prettier/prettier-vscode/issues/888
made default 100 instead of default 80 
![image](https://user-images.githubusercontent.com/76821980/108940923-cd90a200-7619-11eb-9e01-3d8495219a4e.png)

I have googled other issues but I am not sure if they apply to my issue

**Comment by HunterJuneau** [comment, marked as answer]
The red lines you highlighted seem to show proper formatting. The one issue that I see there and that I've been dealing with as well is it putting `new Promise((resolve, reject) => {` onto a new line. The only way to avoid that I know is to only select the content of the Promise (not the Promise declaration or closing brace) and right-click and select `Format Selection`.

**Comment by TrinityChristiana** [comment]
I've had situations like this and had to turn off my prettier or just not autoformat the file.  I actually had to turn mine off for the hackathon this weekend because it was formatting my code in a way that made ESLint upset.

**Comment by drteresavasquez** [comment]
@JuanDavila1101 Hunter is right. 

I would also say there are 3 ways to resolve this and why I LOVE eslint. 1. use this as an opportunity to learn how to format properly as you code so you won't need prettier (E13 will tell you that their coding skills got so much better by following the rules of the linter rather than trying to code and use a formatter) and/or 2. Disable prettier and/or 3. Set the Print width on prettier to 600 (that is how we have the eslint set) so that may help with the link breaks. 

Prettier is perfect for when you aren't using code styles to keep things looking good, but our ESlint config will force you to make things looks right and you will get better and faster at it as you continue on through the next 6 months. 😄
